src/app/ui/editor/moving_pixels_state.cpp

```diff
@@ -501,9 +506,25 @@ void MovingPixelsState::onBeforeCommandExecution(CommandExecutionEvent& ev)
     return;
 
   // We don't need to drop the pixels if a MoveMaskCommand of Content is executed.
-  if (MoveMaskCommand* moveMaskCmd = dynamic_cast<MoveMaskCommand*>(ev.command())) {
+  if (MoveMaskCommand* moveMaskCmd = dynamic_cast<MoveMaskCommand*>(command)) {
     if (moveMaskCmd->getTarget() == MoveMaskCommand::Content) {
       // Do not drop pixels
+      // Verify Shift condition of the MoveMaskCommand (i.e. wrap = true)
+      if (moveMaskCmd->isWrap()) {
+        int dx = 0;
+        int dy = 0;
+        MoveThing::Direction direction = moveMaskCmd->getDirection();
```

Comment:
I think that instead of getting the direction we can the delta directly:
```c++
gfx::Point delta = moveMaskCmd->getDelta(UIContext::instance());
m_pixelsMovement->shift(delta.x, delta.y);
```